docs/DeveloperGuide.md

```diff
@@ -543,30 +543,55 @@ The application provides users with two different methods of entering or editing
 * Using the pop-up text box implemented in this feature
 * Adding the remark through the [add feature](#add-feature)
 
+The remark mechanism will be facilitated by a pop-up text box.
+They can do this by simply omitting the optional `REMARK` parameter, which will cause the pop-up text box to appear.
+This will allow users to format their remarks however they like, rather than being restricted to a single line in the command line.
+Users can use the `CTRL` + `S` combination to close the remark box without using the mouse to click on the "X", thereby
+staying through to the goal of catering to fast typists.
+
+However, should the user prefer it, they can also directly input their remarks into the pop-up text box
+by specifying the optional `REMARK` parameter.
+
+Currently, to differentiate between an empty remark specified from the text box and the `remark INDEX` command to open
+the text box, empty, or whitespace only remarks are represented by the null character `\0`. This means that if the
+user somehow types only a single `\0` character into the text box, it can be seen as a bug as it will be cleared 
+upon closing the text box. However, we do not consider this to be a legitimate use of the program, and see no use in
```

Comment:
Eh we might need to say it we'll fix it in the planned enhancement (or maybe we alrdy have)